packages/components/data-table/src/cell.js

```diff
@@ -18,46 +18,98 @@ import {
   RowExpandCollapseButton,
   HeaderCellInnerWrapper,
 } from './cell.styles';
+import Resizer from './column-resizer';
+
+import ColumnResizingContext from './column-resizing-context';
+
+const HeaderCellWrapper = (props) => {
+  const columnResizingReducer = React.useContext(ColumnResizingContext);
+  const headerRef = React.useRef(null);
+
+  const onStartResizing = (event) => {
+    columnResizingReducer.startResizing(headerRef, event);
+  };
+
+  const onDrag = (event) =>
+    columnResizingReducer.onDragResizing(event, headerRef.current.cellIndex);
+
+  const onDragEnd = () => {
+    columnResizingReducer.finishResizing();
+
+    window.removeEventListener('mousemove', onDrag);
+    window.removeEventListener('mouseup', onDragEnd);
```

Comment:
Maybe for the future we can see how to wrap this into React hooks. Not sure myself now but this adding and removing could maybe be.